pkg/radix/tree.go

```diff
@@ -190,18 +187,23 @@ func (t *Tree) Insert(s string, v int) (int, bool) {
 
 		// No edge, create one
 		if n == nil {
+			newNode := &node{
+				leaf: leafNode{
+					key:   byteCopy(s),
```

Comment:
I think you could save an allocation here with something like:
```
buf := make([]byte, len(s)+len(search)
copy(buf[:len(s)], s)
keyCopy := buf[:len(s)]
copy(len(s):], search)
searchCopy := buf[len(s):]
```

Reply:
nice!